### PR TITLE
chore: bump `@tanstack/react-query` dependency

### PR DESCRIPTION
**Pull request by juliusmarminge**
required for #5913

### CODE REVIEW

**Comment by juliusmarminge** [comment]
bumping to latest seems to cause a ts error on `useSuspenseQueries` so trying to bisect what version broke

**Comment by juliusmarminge** [comment]
seems like https://github.com/TanStack/query/pull/8082 broke it

**Comment by juliusmarminge** [comment]
opted for `queryFn as any` for now. it seems like that's what they're doing in core now too

CleanShot 2024-10-13 at 21 29 32@2x

**Comment by juliusmarminge** [comment]
> opted for `queryFn as any` for now. it seems like that's what they're doing in core now too
> 
> CleanShot 2024-10-13 at 21 29 32@2x

nvm found the actual bug https://github.com/trpc/trpc/pull/6118/commits/94adf64b2cf2c0509b7e25bf20ab67f6078de046

File: packages/react-query/src/shared/hooks/createHooksInternal.tsx [hunk 1]
```diff
@@ -606,6 +606,7 @@ export function createRootHooks<
       {
         queries: queries.map((query) => ({
           ...query,
+          queryFn: query.queryFn,
```
Comment (juliusmarminge) [inline packages/react-query/src/shared/hooks/createHooksInternal.tsx:609]: weirdly enough it doesn't pass without this explicit access on queryFn:

CleanShot 2024-10-13 at 21 40 35@2x

**Review by KATT — APPROVED** [review]

File: packages/react-query/src/shared/hooks/createHooksInternal.tsx (hunk 1 above)
Comment (juliusmarminge) [reply packages/react-query/src/shared/hooks/createHooksInternal.tsx:609]: this is absurd... i think the upstream changes in https://github.com/TanStack/query/pull/8082 is tangling with the type requirement of `queryFn`. 

I struggle to get this to pass in `trpc.useSupenseQueries` when it looks identical to a working `trpc.useQueries`

CleanShot 2024-10-13 at 23 58 09

Comment (juliusmarminge) [reply packages/react-query/src/shared/hooks/createHooksInternal.tsx:609]: it really wants that queryFn here lol...

CleanShot 2024-10-14 at 00 14 08

CleanShot 2024-10-14 at 00 14 42

Comment (juliusmarminge) [reply packages/react-query/src/shared/hooks/createHooksInternal.tsx:609]: Made an upstream PR : https://github.com/TanStack/query/pull/8176

File: patches/@tanstack__react-query@5.59.13.patch
Comment (juliusmarminge) [inline patches/@tanstack__react-query@5.59.13.patch:1]: https://github.com/TanStack/query/pull/8176

File: packages/react-query/src/utils/inferReactQueryProcedure.ts [hunk 2]
```diff
@@ -30,7 +30,7 @@ export type InferQueryOptions<
     TRPCClientErrorLike<TRoot>,
     TData
   >,
-  'select'
+  'select' | 'queryFn'
```
Comment (juliusmarminge) [inline packages/react-query/src/utils/inferReactQueryProcedure.ts:33]: cc @Nick-Lucas - this seems to do it but I'm ngl I don't know if there're broader implications by this change 😅

**Review by KATT — DISMISSED** [review]

File: packages/next/package.json [hunk 3]
```diff
@@ -73,7 +73,7 @@
   ],
   "dependencies": {},
   "peerDependencies": {
-    "@tanstack/react-query": "^5.59.15",
+    "@tanstack/react-query": "^5.54.1",
```
Comment (juliusmarminge) [inline packages/next/package.json]: keeping the peer dep as-is for now

will bump it as part of #5913 where it's actually required.

**Pull request merged**